### PR TITLE
Speed up and using mask image

### PR DESCRIPTION
**Pull request by iron316**
## Changes
- I add new option that load from mask image (i.e. `IMAGE_DIR/trainA_mask` ) 
- I add new script that create mask using LFLSeg module

## What is advantage?
- Learning Leaf GAN is about 3 times faster
- We can leaf masking from other model (i.e. DPT)

### CODE REVIEW

**Comment by iron316** [comment]
I want to review for last check.

File: data/unaligned_masked_dataset.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    def __call__(self, img, mask):
+        if self.grayscale is not None:
+            img = self.grayscale(img)
+            mask = self.grayscale(mask)
+        img = self.to_tensor(self.resize(img))
+        img = self.normalize(img)
+        mask = self.to_tensor(self.resize(mask))
+        # convert float to binary
+        mask = (mask > 0.5).float()
```
Comment (iron316) [inline data/unaligned_masked_dataset.py]: I checked this values.
When create mask, value of mask is binary (0 or 255). But when load mask image, value of mask is not binary(unique values of mask as follow).
```
(Pdb) set(A_mask.numpy().flatten().tolist())
{0.0, 0.007843137718737125, 0.003921568859368563, 0.0235294122248888, 0.01568627543747425, 0.0470588244497776, 0.05882352963089943, 0.03529411926865578, 0.039
21568766236305, 0.0313725508749485, 0.019607843831181526, 0.04313725605607033, 0.027450980618596077, 0.05098039284348488, 0.0117647061124444, 0.05490196123719
2154, 0.9490196108818054, 0.95686274766922, 0.9254902005195618, 0.0784313753247261, 0.07058823853731155, 0.11372549086809158, 0.9803921580314636, 0.9843137264
251709, 0.9725490212440491, 0.9607843160629272, 0.07450980693101883, 1.0, 0.08627451211214066, 0.0941176488995552, 0.08235294371843338, 0.9960784316062927, 0.
9882352948188782, 0.8549019694328308, 0.9411764740943909, 0.06666667014360428, 0.8588235378265381, 0.9450980424880981, 0.9333333373069763, 0.9921568632125854,
 0.9764705896377563, 0.9372549057006836, 0.9215686321258545, 0.9647058844566345, 0.9686274528503418, 0.8823529481887817, 0.9529411792755127, 0.866666674613952
6, 0.929411768913269, 0.8745098114013672, 0.062745101749897, 0.10588235408067703, 0.09019608050584793, 0.10980392247438431, 0.11764705926179886, 0.89019608497
61963, 0.8941176533699036, 0.1411764770746231, 0.13333334028720856, 0.14509804546833038, 0.9058823585510254, 0.9098039269447327, 0.9137254953384399, 0.9176470
637321472}
```

 Sorry, I can't understand why this occur. 
Thus, I introduce line that convert float to binary.
After this change, values of mask is binary.

Comment (huuquan1994) [inline data/unaligned_masked_dataset.py]: Don't worry. Your code is great.
I think this line should be `mask = (mask >= 0.5).float()` kana.

File: prepare_mask.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            num_workers=1,
+        )
+        heatmaps = get_heatmap(netLFLSeg, loader, device)
+        save_heatmap(heatmaps, paths, out_dir, opt.image_size, opt.threshold)
+
+    print("done")
+
+
+if __name__ == "__main__":
+    main()
```
Comment (huuquan1994) [inline prepare_mask.py]: Sugoi desu. Can you move this one to the `util` folder? Thank you so much!

**Review by huuquan1994 — COMMENTED** [review]
I put 2 minor comments. Onegaishimasu!

**Review by huuquan1994 — APPROVED** [review]
Everything is ready to merge!

**Pull request merged**